feat(sonarr)!: add profiles methods

Fuochi opened the pull request:
@davidnewhall Another draft with another idea (that can be easily put aside).
What do you think of returning the entire object instead of its ID (e.g. *QualityProfile over int64)?
This would actually be a quite big breaking change, but it would be nice to have the whole response instead of just its ID to avoid making a get after almost every post/put.

Of course, if you like this approach, I could help you in updating other methods. But feel free to reject it!

davidnewhall reviewed (changes requested):
> What do you think of returning the entire object instead of its ID (e.g. *QualityProfile over int64)?

I'm okay with breaking changes that make sense. When I wrote that method it didn't make any sense to me that someone needed that whole object back, but after using this library more and more I realize it's useful and returning it is the right thing to do.

My request for this contribution is to split it up. Move the breaking changes and the new features into different branches/PRs.  I'm adding you to this repo with write access, so you can clone and push branches. Might make your development a bit easier.

Fuochi commented:
> > What do you think of returning the entire object instead of its ID (e.g. *QualityProfile over int64)?
> 
> I'm okay with breaking changes that make sense. When I wrote that method it didn't make any sense to me that someone needed that whole object back, but after using this library more and more I realize it's useful and returning it is the right thing to do.
> 
> My request for this contribution is to split it up. Move the breaking changes and the new features into different branches/PRs. I'm adding you to this repo with write access, so you can clone and push branches. Might make your development a bit easier.

Great, thank you, as soon as I have some time I'll split it up.

Fuochi commented:
Hi @davidnewhall,
Just rebased and renamed this PR to be aligned with the scope.
I also tried to put in place all the writing style that we agreed on in #40.

Full list of changes in this PR are then:

- ! change output of languageprofile, qualityprofile, and releaseprofile from ID to struct
- add DeleteReleaseProfile method
- add DeleteQualityProfile method
- add GetLanguageProfile, AddLanguageProfile, UpdateLanguageProfile, and DeleteLanguageProfile method

I haven't split up the new features of languageprofile, qualityprofile, and releaseprofile from the output changes. If you prefer, I can do that as well.

davidnewhall commented on sonarr/releaseprofile.go:
```diff
@@ -10,55 +10,103 @@ import (
 
-	_, err := s.GetInto(ctx, "v3/releaseProfile", nil, &profiles)
-	if err != nil {
+	id := strconv.Itoa(profileID)
+	if _, err := s.GetInto(ctx, releaseProfileBP+"/"+id, nil, &output); err != nil {
```
Something that occurred to me is that we can and probably should use `path.Join()` for these. What do you think about this?
```suggestion
	uri := path.Join(releaseProfileBP, strconv.Itoa(profileID))
	if _, err := s.GetInto(ctx, uri, nil, &output); err != nil {
```

Fuochi replied:
I like this, just changed!
I also changed this behaviour in the other profile related files, to keep them consistent

davidnewhall commented on sonarr/languageprofile.go:
```diff
@@ -0,0 +1,115 @@
+}
+
+// Define Base Path for Language Profile calls.
+const languageProfileBP = APIver + "/languageProfile"
```
I like this idea, and thought about something similar. Lets move the suffix to a prefix. Makes auto completion a lot handier for me. 
```suggestion
const bpLanguageProfile = path.Join(APIver, "languageProfile")
```

(don't click accept on this suggestion since there are other changes needed)

davidnewhall replied:
using `path.join` here won't work. oh well.

davidnewhall reviewed (changes requested):
Looks great, just a couple comments.

davidnewhall reviewed (approved):
This looks good, thank you!

Fuochi commented:
Thanks @davidnewhall.
Last time I merged with the merging commit, but I later noticed that you usually don't do that.
Which approach do you prefer?

davidnewhall commented:
I usually squash and merge on this library. You can do a merge commit though. Either works for me.
